src/net/sourceforge/kolmafia/session/LocketManager.java

```diff
@@ -0,0 +1,148 @@
+package net.sourceforge.kolmafia.session;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.Modifiers;
+import net.sourceforge.kolmafia.MonsterData;
+import net.sourceforge.kolmafia.RequestThread;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.persistence.DebugDatabase;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.request.GenericRequest;
+
+public class LocketManager {
+  private static final Set<Integer> knownMonsters = new TreeSet<>();
+  private static final Set<Integer> foughtMonsters = new TreeSet<>();
+  private static final Pattern REMINISCABLE_MONSTER = Pattern.compile("<option value=\"(\\d+)\"");
+  private static final Set<String> CONSTANT_MODS =
+      Set.of("HP Regen Min", "HP Regen Max", "MP Regen Min", "MP Regen Max", "Single Equip");
+
+  private static void parseFoughtMonsters() {
+    foughtMonsters.clear();
+
+    Arrays.stream(Preferences.getString("_locketMonstersFought").split(","))
+        .filter(Predicate.not(String::isBlank))
+        .map(Integer::parseInt)
```

Comment:
We have a convention of not assuming that a property has been only modified by KoLmafia itself. What if the user has modified this setting and stuck in cruft?

Perhaps another filter - StringUtilities.isNumeric - say.

Better than throwing a NumberFormatException.